app/AppKernel.php

```diff
@@ -81,15 +81,34 @@ public function registerBundles()
     /**
      * {@inheritdoc}
      */
-    public function reboot($warmupDir)
+    public function boot()
```

Comment:
What is the relationship between this change and the PR scope?

Reply:
Now that integration tests run without the process isolation they all run in the same process But when you use a `KernelTestCase` it actually boots and shutdown a kernel in `setUp` and `tearDown` classes

But since we store references of the kernel in some places they needed to be cleared or else the kernel from the previous test was used which resulted in weird unexpected bugs

I tried to explain this in the comment of `cleanKernelReferences` but maybe it's not clear enough?